pkg/query/iter.go

```diff
@@ -159,7 +159,7 @@ func (s *chunkSeries) Labels() labels.Labels {
 	return s.lset
 }
 
-func (s *chunkSeries) Iterator() chunkenc.Iterator {
+func (s *chunkSeries) Iterator(iterator chunkenc.Iterator) chunkenc.Iterator {
```

Comment:
Nit: since we're not reusing the iterators, I'd either used unnamed parameter everywhere or preferably `_ chunkenc.Iterator` to indicate it is unused